paddle/phi/kernels/onednn/reduce_max_kernel.cc

```diff
@@ -0,0 +1,66 @@
+/* Copyright (c) 2022 PaddlePaddle Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License. */
+
+#include "paddle/phi/core/kernel_registry.h"
+#include "paddle/phi/kernels/onednn/reduce_kernel.h"
+
+namespace phi {
+
+template <typename T, typename Context>
+void ReduceMaxKernel(const Context& dev_ctx,
```

Comment:
ReduceMaxKernel need to call ReduceMaxRawKernel to implement, and ReduceMaxKernel is device-independent kernel, for it, we only need to add registration in the last of file `kernels/reduce_max_kernel.cc`, here we only need to implement ReduceMaxRawKernel